From 8572b2205bd02bfe3a6fe609833f8450b8c263d7 Mon Sep 17 00:00:00 2001
From: Patrick Mooney <pmooney@pfmooney.com>
Date: Tue, 20 Dec 2016 18:35:03 +0000
Subject: [PATCH] OS-5867 header install racy in devfsadm build Reviewed by:
 Robert Mustacchi <rm@joyent.com> Reviewed by: Ryan Zezeski
 <ryan.zeseski@joyent.com> Approved by: Ryan Zezeski <ryan.zeseski@joyent.com>

---
 usr/src/Makefile              | 1 +
 usr/src/cmd/devfsadm/Makefile | 7 +++++--
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/usr/src/Makefile b/usr/src/Makefile
index 0f0fb09629..5fe34ddd48 100644
--- a/usr/src/Makefile
+++ b/usr/src/Makefile
@@ -220,6 +220,7 @@ ucbheaders: FRC
 	@cd ucbhead; pwd; $(MAKE) install_h
 
 cmdheaders: FRC
+	@cd cmd/devfsadm; pwd; $(MAKE) install_h
 	@cd cmd/fm; pwd; $(MAKE) install_h
 	@cd cmd/mdb; pwd; $(MAKE) install_h
 
diff --git a/usr/src/cmd/devfsadm/Makefile b/usr/src/cmd/devfsadm/Makefile
index 7b2409d131..aec7a8ec8b 100644
--- a/usr/src/cmd/devfsadm/Makefile
+++ b/usr/src/cmd/devfsadm/Makefile
@@ -20,6 +20,7 @@
 #
 # Copyright 2009 Sun Microsystems, Inc.  All rights reserved.
 # Use is subject to license terms.
+# Copyright 2016 Joyent, Inc.
 #
 # cmd/devfsadm/Makefile
 #
@@ -61,8 +62,10 @@ clean lint _msg:	$(SUBDIRS)
 clobber: $(SUBDIRS)
 	$(RM) $(ROOTETCDEVFILES)
 
-install: $(SUBDIRS) $(ROOTETCDEFAULTFILES) $(ROOTETCDEVFILES) \
-	$(ROOTUSRINCLUDEFILES)
+install_h: $(ROOTUSRINCLUDEFILES)
+
+install: $(SUBDIRS) install_h $(ROOTETCDEFAULTFILES) $(ROOTETCDEVFILES)
+
 
 $(ROOTETCDEV):
 	$(INS.dir)
-- 
2.21.0

